internal/engine/pod.go

```diff
@@ -148,24 +148,21 @@ func matchPodChangeToManifest(state *store.EngineState, action k8swatch.PodChang
 	return mt
 }
 
-// Checks the runtime state if we're already tracking this pod.
-// If not, AND if the pod matches the current deploy, create a new tracking object.
-// Returns a store.Pod that the caller can mutate, and true
-// if this is the first time we've seen this pod.
-func maybeTrackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) (*v1alpha1.Pod, bool) {
+// trackPod adds a Pod to the RuntimeState for a resource.
+//
+// If there are currently tracked Pods and the new Pod's ancestor UID differs from theirs,
+// the tracked Pods will be cleared and the ancestor UID updated. Otherwise, the Pod will
+// be upserted and the prior version returned.
+//
+// A mutable reference to the Pod to be tracked is stored so that further changes to it
+// can be made based on the diff from the returned prior version. The caller should *not*
+// hold their reference beyond the action handler.
+func trackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) *v1alpha1.Pod {
```

Comment:
tbh I've always had trouble reasoning about what this func returns, so if you have naming thoughts that might clarify, I'm v in favor. `trackPodAndReturnExisting` (or maybe better yet `trackPodAndReturnPrevious`) is... more readable, maybe annoyingly so? idk.